File: src/main/java/com/ufrn/imd/web2/projeto01/livros/models/Product.java
package com.ufrn.imd.web2.projeto01.livros.models;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

@Entity
@Table(name="product")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer id;

    @Column
    private Integer creator;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name="book_id")
    private Book book;
    
    @Column
    private Integer quantity;

    @Column
    private Float price;

    @ManyToOne
    @JoinColumn(name = "bookstore_id")
    private Bookstore bookstore;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Integer getCreator() {
        return creator;
    }

    public void setCreator(Integer creator) {
        this.creator = creator;
    }

    public Book getBook() {
        return book;
    }

    public void setBook(Book book) {
        this.book = book;
    }

    public Integer getQuantity() {
        return quantity;
    }

    public void setQuantity(Integer quantity) {
        this.quantity = quantity;
    }

    public Float getPrice() {
        return price;
    }

    public void setPrice(Float price) {
        this.price = price;
    }

    public Bookstore getBookstore() {
        return bookstore;
    }

    public void setBookstore(Bookstore bookstore) {
        this.bookstore = bookstore;
    }

    @Override
    public String toString() {
        return "Product [id=" + id + ", creator=" + creator + ", book=" + book + ", quantity=" + quantity + ", price="
                + price + ", bookstore=" + bookstore + "]";
    }

}

File: book_reposity_app/lib/models/product.dart
import 'dart:ffi';

import 'book_model.dart';
import 'bookstore.dart';

class Product {
  String id;
  Book book;
  Float price;
  int quantity;
  Bookstore bookstore;

  Product(
      {required this.id,
      required this.book,
      required this.price,
      required this.quantity,
      required this.bookstore});

  factory Product.fromJson(Map json) {
    Book newBook = Book.fromJson(json["book"]);

    Bookstore newBookstore = Bookstore.fromJson(json["bookstore"]);

    return Product(
        id: json["id"].toString(),
        book: newBook,
        price: json["price"],
        quantity: json["quantity"],
        bookstore: newBookstore);
  }

  Map<String, dynamic> toJson(int bookId, int bookstoreId) {
    return {
      "bookId": bookId,
      "price": price,
      "quantity": quantity,
      "booktoreId": bookstoreId
    };
  }
}
